src/common/lib/util/eventemitter.ts

```diff
@@ -286,32 +286,18 @@ class EventEmitter {
   }
 
   /**
-   * Private API
-   *
    * Listen for a single occurrence of a state event and fire immediately if currentState matches targetState
    * @param targetState the name of the state event to listen to
    * @param currentState the name of the current state of this object
-   * @param listener the listener to be called
-   * @param listenerArgs
    */
-  whenState(targetState: string, currentState: string, listener: Function, ...listenerArgs: unknown[]) {
-    const eventThis = { event: targetState };
-
+  async whenState(targetState: string, currentState: string) {
     if (typeof targetState !== 'string' || typeof currentState !== 'string') {
       throw 'whenState requires a valid event String argument';
```

Comment:
should change to `throw new Error(...)` instead of throwing a string.

also message is a bit misleading using the term `event` here.
maybe like this is better:
`whenState requires a valid state String argument`

Reply:
Good catch, was like that already but will change.